root/static/scripts/edit/URLCleanup.js

```diff
@@ -4291,7 +4291,10 @@ const CLEANUPS: CleanupEntries = {
     },
   },
   'vimeo': {
-    match: [new RegExp('^(https?://)?([^/]+\\.)?(vimeo\\.com/)', 'i')],
+    match: [new RegExp(
+      '^(https?://)?([^/]+\\.)?vimeo\\.com/(?!ondemand)',
+      'i',
+    )],
```

Comment:
It should not match `/store/ondemand` path either, for example:
* `https://vimeo.com/store/ondemand/buy/91410`
* `https://vimeo.com/store/ondemand/rent/91410`

Reply:
Yeah. But I guess there's no way to turn these back into the original non-store link, so we should reject these under the ondemand regexp and ask people to add the main page instead.

Reply:
> I guess there's no way to turn these back into the original non-store link, so we should reject these under the ondemand regexp

Yes.

> ask people to add the main page instead.

Just mentioning the expected URL format should be fine. It can be a generic error message for other websites too.